app/flatpak-builtins-build-bundle.c

```diff
@@ -177,6 +173,10 @@ iterate_bundle_icons (GFile                     *root,
   g_autoptr(GFile) icons_dir =
     g_file_resolve_relative_path (root,
                                   "files/share/app-info/icons/flatpak");
+  if (!g_file_test (g_file_peek_path (icons_dir), G_FILE_TEST_IS_DIR)) {
+    icons_dir = g_file_resolve_relative_path (root,
```

Comment:
Sorry I missed this before merge but you leak `icons_dir` here.

Reply:
I'm not to comfortable with c. what would be the function to not leak icons_dir?

Reply:
I've fixed this, I'll open a pr with the fix